internal/helper/diag.go

```diff
@@ -53,8 +53,8 @@ func AddDiagnosticOnError(diag *diag.Diagnostics, summary string, err error) boo
 }
 
 func (d DetailedErrorDiagnostic) Detail() string {
+	response := fmt.Sprintf("ID: %s\n", d.e.ID)
```

Comment:
We're doing some confusing things with looping over the Details of the error to generate the message, and it's otherwise not super consistent as far as I can tell. Super open to other ideas of how to format this.